File: src/main/java/wraith/fabricaeexnihilo/modules/infested/SilkWormItem.java
package wraith.fabricaeexnihilo.modules.infested;

import net.fabricmc.fabric.api.item.v1.FabricItemSettings;
import net.minecraft.item.Item;
import net.minecraft.item.ItemUsageContext;
import net.minecraft.util.ActionResult;

public class SilkWormItem extends Item {

    public SilkWormItem(FabricItemSettings settings) {
        super(settings);
    }

    @Override
    public ActionResult useOnBlock(ItemUsageContext context) {
        var result = InfestedHelper.tryToInfest(context.getWorld(), context.getBlockPos());
        if (result == ActionResult.SUCCESS && (context.getPlayer() == null || !context.getPlayer().isCreative())) {
            context.getStack().decrement(1);
        }
        return result;
    }

}

File: src/main/kotlin/exnihilofabrico/modules/infested/SilkWormItem.kt
package exnihilofabrico.modules.infested

import exnihilofabrico.modules.base.BaseItem
import net.minecraft.item.ItemUsageContext
import net.minecraft.util.ActionResult

class SilkWormItem(settings: Settings): BaseItem(settings) {
    override fun useOnBlock(context: ItemUsageContext?): ActionResult {
        if(context==null || context.world.isClient)
            return super.useOnBlock(context)

        val result = InfestedHelper.tryToInfest(context.world, context.blockPos)

        if(result == ActionResult.SUCCESS && context.player?.isCreative != true)
            context.stack.decrement(1)

        return result
    }
}